base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+immutable Transpose{T,S<:AbstractMatrix} <: AbstractMatrix{T}
+    data::S
+    conjugated::Bool
+end
```

Comment:
What I have been wondering about matrix transpose (and the same applies to vector transpose as implemented by @andyferris ): How many cases are there where you actually want to normal (not conjugated transpose) of a complex matrix. What if we start by only having a wrapper for `ctranspose`, which just always complex conjugates. For real numbers, the compiler would optimize away the `conj` call. We are anyway forcing people to use `'` as a mathematical operation, by making it act recursively and pointing them to `permutedims` for reshuffling dimensions of a matrix which is filled with custom data.

So the question is, how many math applications of the non-conjugated transpose of a complex matrix are there. I know BLAS supports this, but is that a proper justification?